op-batcher/batcher/espresso.go

```diff
@@ -21,43 +22,529 @@ import (
+		transaction: job,
+	}
+}
+
+// handleTransactionSubmitJobResponse is a function that is meant to be run in a
+// goroutine.
+//
+// It handles the responses from the submit transaction jobs.  It will
+// determine if the transaction was successfully submitted to Espresso, and
+// if not, it will retry the transaction.  If the transaction was successfully
+// submitted, it will then submit a job to the verify receipt job queue to
+// verify the receipt of the transaction.
+func (s *espressoTransactionSubmitter) handleTransactionSubmitJobResponse() {
 	for {
+		var jobResp espressoSubmitTransactionJobResponse
+		var ok bool
+
 		select {
-		case <-ticker.C:
-			_, err := l.Espresso.FetchTransactionByHash(ctx, txHash)
-			if err == nil {
-				return nil
+		case <-s.ctx.Done():
+			return
+		case jobResp, ok = <-s.submitRespQueue:
+			if !ok {
+				// Our channel is closed, and we are done
+				return
+			}
+		}
+
+		// TODO: Evaluate the specific error type, and determine if we
+		// should retry
+		if jobResp.err != nil {
+			s.submitJobQueue <- jobResp.job
+			continue
+		}
+
+		verifyJob := espressoVerifyReceiptJob{
+			start:       time.Now(),
+			transaction: jobResp.job,
+			hash:        jobResp.hash,
+		}
+
+		select {
+		case <-s.ctx.Done():
+			return
+		// Move to verifying the receipt
+		case s.verifyReceiptJobQueue <- verifyJob:
+		}
+	}
+}
+
+// VERIFY_RECEIPT_TIMEOUT is the amount of time we will wait for a receipt to
+// be verified before we requeue the job for another attempt.
+const VERIFY_RECEIPT_TIMEOUT = 4 * time.Second
+
+// VERIFY_RECEIPT_RETRY_DELAY is the amount of time we will wait before
+// retrying a job that failed to verify the receipt.
+const VERIFY_RECEIPT_RETRY_DELAY = 100 * time.Millisecond
+
+// handleVerifyReceiptJobResponse is a function that is meant to be run in a
+// goroutine.
+//
+// This function handles responses from the verify receipt job queue.  It will
+// check the results for any errors, and if there are any errors that are
+// applicable to retry, it will requeue the job for another attempt.
+// If the the job is successful, no further processing is needed and it is
+// considered complete.
+// If the job has taken too long to verify, then it will re-submit the job
+// back to the submit transaction queue for another attempt.
+//
+// NOTE: This function currently will loop forever if the transaction is
+// never going to be available.
+//
+// TODO: we need to put some sensible limits on the number of times we will
+// retry a job, depending on the type of the error we received.
+func (s *espressoTransactionSubmitter) handleVerifyReceiptJobResponse() {
+	for {
+		var jobResp espressoVerifyReceiptJobResponse
+		var ok bool
+
+		select {
+		case <-s.ctx.Done():
+			return
+		case jobResp, ok = <-s.verifyReceiptRespQueue:
+			if !ok {
+				// Our channel is closed, and we are done
+				return
+			}
+		}
+
+		// TODO: Evaluate the specific error type, and determine if we
```

Comment:
Are we ok leaving this TODO for after the audit?

Reply:
I am alright with this.  I was just attempting to feature match what @QuentinI had already done.